homeassistant/components/eq3btsmart/binary_sensor.py

```diff
@@ -0,0 +1,135 @@
+"""Platform for eQ-3 binary sensor entities."""
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    DOMAIN,
+    ENTITY_NAME_BATTERY,
+    ENTITY_NAME_BUSY,
+    ENTITY_NAME_CONNECTED,
+    ENTITY_NAME_DST,
+    ENTITY_NAME_WINDOW_OPEN,
+)
+from .entity import Eq3Entity
+from .models import Eq3ConfigEntryData
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the entry."""
+
+    eq3_config_entry: Eq3ConfigEntryData = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities_to_add: list[Entity] = [
+        BatterySensor(eq3_config_entry.eq3_config, eq3_config_entry.thermostat),
+        WindowOpenSensor(eq3_config_entry.eq3_config, eq3_config_entry.thermostat),
+        DSTSensor(eq3_config_entry.eq3_config, eq3_config_entry.thermostat),
+        BusySensor(eq3_config_entry.eq3_config, eq3_config_entry.thermostat),
+        ConnectedSensor(eq3_config_entry.eq3_config, eq3_config_entry.thermostat),
+    ]
+
+    async_add_entities(entities_to_add)
+
+
+class BusySensor(Eq3Entity, BinarySensorEntity):
```

Comment:
Instead of defining all kinds of classes, it would be better to make use of entity descriptions (Look for `BinarySensorEntityDescription` in our codebase for examples).